src/client/app/components/UIOptionsComponent.jsx

```diff
@@ -93,14 +132,50 @@ export default class UIOptionsComponent extends React.Component {
 						<div className="checkbox">
 							<label><input type="checkbox" onChange={this.handleChangeBarStacking} checked={this.props.barStacking} />Bar stacking</label>
 						</div>
-						<p style={labelStyle}>Bar chart interval (days):</p>
-						<Slider min={1} max={365} value={this.state.barDuration} onChange={this.handleBarDurationChange} onChangeComplete={this.handleBarDurationChangeComplete} />
+						<p style={labelStyle}>Bar chart interval:</p>
+						<ToggleButtonGroup
+							type="radio"
+							name="timeSpans"
+							value={this.state.barDuration}
+							onChange={this.handleSpanButton}
+							style={zIndexFix}
+						>
+							<ToggleButton value={1}>Day</ToggleButton>
+							<ToggleButton value={7}>Week</ToggleButton>
+							<ToggleButton value={28}>Month</ToggleButton>
+						</ToggleButtonGroup>
+						<Button name="customToggle" onClick={this.toggleSlider}>Toggle Custom Slider (days)</Button>
+
+						{this.state.showSlider &&
+						<Slider
+							min={1} max={365} value={this.state.barDuration} onChange={this.handleBarDurationChange}
+							onChangeComplete={this.handleBarDurationChangeComplete}
+						/>
+						}
 					</div>
+
+				}
+				{this.props.chartToRender === chartTypes.compare &&
+				<div>
+					<ToggleButtonGroup
+						name="timeSpansCompare"
+						value={this.state.compareInterval}
```

Comment:
`this.state.compareInterval` is undefined

I believe you need:
`const compareTimeInterval = TimeInterval.fromString(this.props.compareInterval);`
and
`value={compareTimeInterval.endTimestamp.diff(compareTimeInterval.startTimestamp, 'days')}`
as well as modifying the ToggleButton values and the corresponding event handler.